rotkehlchen/chain/ethereum/modules/liquity/trove.py

```diff
@@ -0,0 +1,506 @@
+                        balance = total_lusd_trove_balance.amount + realized_trove_lusd_loss.amount
+                        if balance < ZERO:
+                            pnl_balance = total_lusd_trove_balance + realized_trove_lusd_loss
+                            realized_trove_lusd_loss += -pnl_balance
+                            pnl = [AssetBalance(asset=A_LUSD, balance=pnl_balance)]
+                if collateral_change != AssetAmount(ZERO):
+                    if collateral_change < ZERO:
+                        # Withdraw collateral
+                        got_asset = A_ETH
+                        got_balance = Balance(
+                            amount=abs(collateral_change),
+                            usd_value=query_usd_price_or_use_default(
+                                A_ETH,
+                                timestamp,
+                                ZERO,
+                                'Liquity',
+                            ),
+                        )
+                    else:
+                        # Deposit collateral
+                        spent_asset = A_ETH
+                        spent_balance = Balance(
+                            amount=collateral_change,
+                            usd_value=query_usd_price_or_use_default(
+                                A_ETH,
+                                timestamp,
+                                ZERO,
+                                'Liquity',
+                            ),
+                        )
+                if operation in (
+                    TroveOperation.LIQUIDATENORMALMODE,
+                    TroveOperation.LIQUIDATERECOVERYMODE,
+                ):
+                    count_spent_got_cost_basis = True
+                    spent_asset = A_ETH
+                    spent_balance = Balance(
+                        amount=abs(collateral_change),
+                        usd_value=query_usd_price_or_use_default(
+                            A_ETH,
+                            timestamp,
+                            ZERO,
+                            'Liquity',
+                        ),
+                    )
+                    pnl = [AssetBalance(asset=A_ETH, balance=-spent_balance)]
+                event = DefiEvent(
+                    timestamp=Timestamp(change['transaction']['timestamp']),
+                    wrapped_event=change,
+                    event_type=DefiEventType.LIQUITY,
+                    got_asset=got_asset,
+                    got_balance=got_balance,
+                    spent_asset=spent_asset,
+                    spent_balance=spent_balance,
+                    pnl=pnl,
+                    count_spent_got_cost_basis=count_spent_got_cost_basis,
+                    tx_hash=change['transaction']['id'],
+                )
+                events.append(event)
+            except KeyError as e:
+                self.msg_aggregator.add_warning(
```

Comment:
Add also a log warning with the actual entry so that we can inspect the logs to find the offending entry.